staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -128,35 +151,76 @@ type MutableFeatureGate interface {
 	OverrideDefault(name Feature, override bool) error
 }
 
+// MutableVersionedFeatureGate parses and stores flag gates for known features from
+// a string like feature1=true,feature2=false,...
+// MutableVersionedFeatureGate sets options based on the emulated version of the featured gate.
+type MutableVersionedFeatureGate interface {
+	MutableFeatureGate
+	// EmulationVersion returns the version the feature gate is set to emulate.
+	// If set, the feature gate would enable/disable features based on
+	// feature availability and pre-release at the emulated version instead of the binary version.
+	EmulationVersion() *version.Version
+	// SetEmulationVersion overrides the emulationVersion of the feature gate.
+	// Otherwise, the emulationVersion will be the same as the binary version.
+	// If set, the feature defaults and availability will be as if the binary is at the emulated version.
+	SetEmulationVersion(emulationVersion *version.Version) error
+	// GetAll returns a copy of the map of known feature names to versioned feature specs.
+	GetAllVersioned() map[Feature]VersionedSpecs
+	// AddVersioned adds versioned feature specs to the featureGate.
+	AddVersioned(features map[Feature]VersionedSpecs) error
+}
+
+// MutableVersionedFeatureGateForTests is a feature gate interface that should only be used in tests.
```

Comment:
Which tests?  Tests in this package or tests in other packages?  `Reset` seems like an odd thing to allow outside thsi package.

Reply:
it is used in `featuregatetesting.SetFeatureGateDuringTest`, which is used in many tests so that changing the `DefaultFeatureGate` does not affect other tests.

Reply:
I don't see `Reset` being used.